torch/_refs/__init__.py

```diff
@@ -1103,38 +1103,123 @@ def bitwise_left_shift(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
+
+    This function takes two tensor-like objects `a` and `b`, and performs an element-wise
+    bitwise right shift operation. The type of the resulting tensor is determined by the
+    `ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT` rule.
+
+    Args:
+        a (TensorLikeType): The first input tensor.
+        b (TensorLikeType): The second input tensor, must have the same shape as `a`.
+
+    Returns:
+        TensorLikeType: A tensor with the same shape as `a` and `b`, where each element
+        is the result of bitwise right shift operation on corresponding elements of
+        `a` and `b`.
+
+    Example:
+        >>> a = np.array([10, 20, 30])
+        >>> b = np.array([1, 2, 3])
+        >>> bitwise_right_shift(a, b)
+        array([5, 5, 3])
+    """
     return prims.shift_right_arithmetic(a, b)
 
 
-# TODO: add docstring
 @_make_elementwise_binary_reference(
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
 def bitwise_xor(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
+    """
+    Perform bitwise XOR operation on two tensors.
+
+    This function takes two tensor-like objects `a` and `b` as input and performs an
+    element-wise bitwise XOR operation on them. The type of the resulting tensor is
+    determined by the `ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT` rule, which
+    promotes the type of the operands to a common type.
+
+    Args:
+        a (TensorLikeType): The first input tensor.
+        b (TensorLikeType): The second input tensor.
+
+    Returns:
+        TensorLikeType: A tensor resulting from the bitwise XOR operation on
+        the input tensors.
+
+    Example:
+        >>> a = np.array([9, 8, 7], dtype=np.int32)
+        >>> b = np.array([1, 2, 3], dtype=np.int32)
+        >>> bitwise_xor(a, b)
+        array([8, 10, 4], dtype=int32)
+    """
     return prims.bitwise_xor(a, b)
 
 
-# TODO: add docstring
 @_make_elementwise_binary_reference(
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
     supports_lhs_python_scalar=False,
 )
 def copysign(
     a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]
 ):
+    """
+    Copy the sign of the second input to the first input.
+
+    This function takes two inputs, `a` and `b`, and returns a tensor with the
+    magnitude of `a` and the sign of `b`. If `b` is a scalar, it is converted to a
+    tensor with the same dtype and device as `a`. If `a` and `b` are both tensors,
+    they must be on the same device.
+
+    Args:
+        a: A tensor or a scalar. The magnitude of the output is based on this input.
+        b: A tensor or a scalar. The sign of the output is based on this input.
+
+    Returns:
+        A tensor with the magnitude of `a` and the sign of `b`.
+
+    Raises:
+        RuntimeError: If `a` and `b` are both tensors but are not on the same device.
+
+    Example:
+        >>> a = torch.tensor([1.0, -2.0, 3.0])
+        >>> b = torch.tensor([-1.0, 2.0, -3.0])
```

Comment:
`b` should be probably different from the result to properly show what the function does.